File: api/src/main/java/com/github/av2/api/model/Headquarters.java
package com.github.av2.api.model;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Schema(description = "Headquarters information")
public class Headquarters {
    @Schema(description = "The unique identifier for the headquarters", required = true)
    private Integer headquartersId;
    
    @Schema(description = "The name of the headquarters", required = true)
    private String name;
    
    @Schema(description = "Description of the headquarters")
    private String description;
    
    @Schema(description = "Physical address of the headquarters")
    private String address;
    
    @Schema(description = "Contact person for the headquarters")
    private String contactPerson;
    
    @Schema(description = "Contact email for the headquarters")
    private String email;
    
    @Schema(description = "Contact phone number for the headquarters")
    private String phone;
}

File: api/src/models/headquarters.ts
/**
 * @swagger
 * components:
 *   schemas:
 *     Headquarters:
 *       type: object
 *       required:
 *         - headquartersId
 *         - name
 *       properties:
 *         headquartersId:
 *           type: integer
 *           description: The unique identifier for the headquarters
 *         name:
 *           type: string
 *           description: The name of the headquarters
 *         address:
 *           type: string
 *           description: Main office address of the headquarters
 *         phone:
 *           type: string
 *           description: Contact phone number for the headquarters
 *         email:
 *           type: string
 *           format: email
 *           description: Contact email for the headquarters
 *         description:
 *           type: string
 *           description: Additional details about the headquarters
 */
export interface Headquarters {
  headquartersId: number;
  name: string;
  description: string;
  address: string;
  contactPerson: string;
  email: string;
  phone: string;
}
